src/app/components/device-selection/device-selection.component.ts

```diff
@@ -0,0 +1,58 @@
+import { Component, OnInit } from '@angular/core';
+import { SpotifyPlayerService } from '../../services/spotify.player.service';
+import { CommonModule } from '@angular/common';
+import { interval, Subscription } from 'rxjs';
+
+@Component({
+  selector: 'app-device-selection',
+  standalone: true,
+  imports: [CommonModule],
+  templateUrl: './device-selection.component.html',
+  styleUrl: './device-selection.component.scss',
+})
+export class DeviceSelectionComponent implements OnInit {
+  devices: any[] = [];
+  private _refreshInterval: Subscription | undefined;
+  private _token: string = localStorage.getItem('access_token') || '';
```

Comment:
Ta logika z tym tokenem jest troche niepoprawna. Widzę że to masz też w innych komponentach.

Token powinien być automatycznie wpisywany w headersach w interceptorze - tam pobierany z localstorage.

A to czy token wgl jest albo czy jest aktywny powinieneś sprwdzać w jakiś najwyższym komponencie np app component i jeśli nie ma to wylogować user, nie możesz dopuścić wgl usera do koponentów innych jeśli nie ma tokenu.

Postaraj sie to zmienić i usuń to przekazywanie tokenu we wszystkich metodach w serwisach i to sprawdzanie czy jest token w tych metodach.